tests/Doctrine/Tests/DBAL/Driver/StatementIteratorTest.php

```diff
@@ -2,39 +2,94 @@
 
 namespace Doctrine\Tests\DBAL\Driver;
 
+use Doctrine\DBAL\Driver\IBMDB2\DB2Statement;
+use Doctrine\DBAL\Driver\Mysqli\MysqliStatement;
+use Doctrine\DBAL\Driver\OCI8\OCI8Statement;
+use Doctrine\DBAL\Driver\SQLAnywhere\SQLAnywhereStatement;
+use Doctrine\DBAL\Driver\SQLSrv\SQLSrvStatement;
 use Doctrine\DBAL\Driver\Statement;
 use Doctrine\DBAL\Driver\StatementIterator;
+use Doctrine\DBAL\Portability\Statement as PortabilityStatement;
+use IteratorAggregate;
+use PHPUnit\Framework\MockObject\MockObject;
+use Traversable;
+use function extension_loaded;
 
 class StatementIteratorTest extends \Doctrine\Tests\DbalTestCase
 {
-    public function testGettingIteratorDoesNotCallFetch()
+    /**
+     * @dataProvider statementProvider()
```

Comment:
First time I see data providers with `()`, is this reliable?

Reply:
Yes, I've been always using them. A function name without parentheses looks awkward.